pyro/distributions/util.py

```diff
@@ -15,7 +15,8 @@
 
 from pyro.util import ignore_jit_warnings
 
-_VALIDATION_ENABLED = False
+_VALIDATION_ENABLED = __debug__
+torch_dist.Distribution.set_default_validate_args(__debug__)
```

Comment:
@neerajprad this line changes the default behavior of `torch.distributions`. It may lead to weird behavior since simply importing Pyro will turn on PyTorch validation. However this still seems like the best option to me.